src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceProvider.cs

```diff
@@ -91,7 +91,20 @@ internal ServiceProvider(ICollection<ServiceDescriptor> serviceDescriptors, Serv
         /// </summary>
         /// <param name="serviceType">The type of the service to get.</param>
         /// <returns>The service that was produced.</returns>
-        public object? GetService(Type serviceType) => GetService(serviceType, Root);
+        public object? GetService(Type serviceType) => GetService(ServiceIdentifier.FromServiceType(serviceType), Root);
+
+        public object? GetKeyedService(Type serviceType, object serviceKey)
+            => GetService(new ServiceIdentifier(serviceKey, serviceType), Root);
```

Comment:
So there are no null checks on the `object serviceKey` and calling `GetKeyedService(serviceType, null)` is equivalent to calling `GetService(serviceType)`?

I actually like that behavior, but then we should mark the `serviceKey` as nullable everywhere in the public APIs. If we don't like that behavior, we can throw an `ArgumentNullException`. Either way, we should test the behaviour.